libp2p-ping/src/lib.rs

```diff
@@ -156,8 +159,19 @@ impl<C> ConnectionUpgrade<C> for Ping
 		let sink_stream = socket.framed(Codec).map(|msg| Message::Received(msg.freeze()));
 		let (sink, stream) = sink_stream.split();
 
+		let remote_addr = if log_enabled!(target: "libp2p-ping", Level::Debug) {
+			Some(remote_addr.clone())
+		} else {
+			None
+		};
```

Comment:
I don't understand. Why are you wrapping `remote_addr` into an `Option`?

Reply:
So that we only clone it if logging is enabled. Not very elegant, but it's the only possible way if I'm not mistaken.

Reply:
oh, I see now. Otherwise the reference would not live long enough :)